packages/js/data/src/products/reducer.ts

```diff
@@ -99,6 +99,27 @@ const reducer: Reducer< ProductState, Actions > = (
 						) ]: payload.error,
 					},
 				};
+			case TYPES.DELETE_PRODUCT_ERROR:
+				return {
+					...state,
+					errors: {
+						...state.errors,
+						[ `delete/${ payload.id }` ]: payload.error,
+					},
+				};
+			case TYPES.DELETE_PRODUCT_SUCCESS:
+				const prData = state.data || {};
+				return {
+					...state,
+					data: {
```

Comment:
Yeah, I have a couple thoughts:
- For deleted products we could use the filter either where we display the list of products or within the selector itself, we did probably want to create a different selector `getProductsWithStatus` in that case, that makes use of `getProducts` under the hood. The [use of `useSelector`](https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/data/src/products/selectors.ts#L51-L55) , does allow the component to be re-rendered if a nested product has been updated (whether that is it's status, title, or anything else). So as long we keep any other filters client side, we should be fine on that front.
 
- Make use of `invalidateResolution`, probably only from the component the deletion or update takes place in and the query as well. I am not sure if there is an easy way for us to make this part of the `delete` and `update` actions. I guess we could query for the specific `id` within `products[queryString].data` and invalidate all of them that contain the `id`. As long as we do this silently I think it is fine (meaning we don't wipe the entire table and show it is refreshing).

- Change our redux store, by removing the `products[queryString].data` part and only having a single list of products. We did have to do some filtering on the client side also. This would cause an issue if we ever wanted to display multiple lists of products on the same page (unless we filter everything, including pagination on the front-end).

I say a combination of the first two should be fine.

Reply:
> Leave it as is and let the consumer filter out deleted products.

I am really thinking that this would be the way to go, it all depends on what our layout will look like for a list of products.

Reply:
Thank you @joshuatf and @louwie17 for your feedback on this. So if it's ok with you I will leave the current handling of the deleted products like it is, to continue analyzing the resolution invalidation and tackle it in a follow-up PR.